Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/internal/angles.h

```diff
@@ -1,119 +0,0 @@
-// Copyright (c) 2016  GeometryFactory (France).
-// All rights reserved.
-//
-// This file is part of CGAL (www.cgal.org).
-//
-// $URL$
-// $Id$
-// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
-//
-// Author(s)     : Mael Rouxel-Labbé
-
-#ifndef CGAL_SURFACE_MESH_PARAMETERIZATION_ANGLES_H
-#define CGAL_SURFACE_MESH_PARAMETERIZATION_ANGLES_H
-
-#include <CGAL/license/Surface_mesh_parameterization.h>
-
-#include <CGAL/Kernel/global_functions.h>
-#include <CGAL/number_type_config.h>
-
-#include <cmath>
-
-namespace CGAL {
-
-namespace Surface_mesh_parameterization {
-
-namespace internal {
-
-//                                                                           ->  ->
-// Returns the cotangent of the corner (P,Q,R) (i.e. the cotan of the angle (QP, QR) ).
-template<typename K>
-typename K::FT cotangent(const typename K::Point_3& P,
-                         const typename K::Point_3& Q,
-                         const typename K::Point_3& R)
-{
-  typedef typename K::FT                   NT;
-  typedef typename K::Vector_3             Vector_3;
-
-  Vector_3 u = P - Q;
-  Vector_3 v = R - Q;
-  NT dot = (u * v);
-  Vector_3 cross_vector = CGAL::cross_product(u, v);
-  NT cross_norm = CGAL::sqrt(cross_vector * cross_vector);
-  if(cross_norm != NT(0))
-    return (dot / cross_norm);
-  else
-    return 0; // undefined
-}
-
-//                                                                       ->  ->
-// Returns the tangent of the corner (P,Q,R) (i.e. the tangent of angle (QP, QR) ).
-template<typename K>
-typename K::FT tangent(const typename K::Point_3& P,
-                       const typename K::Point_3& Q,
-                       const typename K::Point_3& R)
-{
-  typedef typename K::FT                 NT;
-  typedef typename K::Vector_3           Vector_3;
-
-  Vector_3 u = P - Q;
-  Vector_3 v = R - Q;
-  NT dot = (u * v);
-  Vector_3 cross_vector = CGAL::cross_product(u, v);
-  NT cross_norm = CGAL::sqrt(cross_vector * cross_vector);
-  if(dot != NT(0))
-    return (cross_norm / dot);
-  else
-    return 0; // undefined
-}
-
-// Fixes the sine to be within [-1;1].
-template<typename K>
-typename K::FT fix_sine(typename K::FT sine)
```

Comment:
That clamping function mattered for the robustness of some methods, is there a similar mechanism in what you replaced the angle computation with?

Reply:
Yes, I sort of checked it. Without `fix_sine`, certain tests did not pass. In the new version, they pass. I do not use exactly this function to fix sines. However, some weights use different formulas, which avoid these problems, and in some other weights, where the angles are required, I also truncate if necessary. To be honest, I did not find a way to check exactly if the new weights use truncations at the same places where old weights did. I simply tested intermediate output values where possible and checked if the test suite passes in this case.